cmd/waku/rest/relay.go

```diff
@@ -187,11 +192,23 @@ func (d *RelayService) postV1Message(w http.ResponseWriter, r *http.Request) {
 
 	var err error
 	if topic == "" {
-		_, err = d.node.Relay().Publish(r.Context(), message)
-	} else {
-		_, err = d.node.Relay().PublishToTopic(r.Context(), message, strings.Replace(topic, "\n", "", -1))
+		topic = relay.DefaultWakuTopic
 	}
 
+	if !d.node.Relay().IsSubscribed(topic) {
+		writeErrOrResponse(w, errors.New("not subscribed to pubsubTopic"), nil)
+		return
+	}
+
+	if node.SupportsRLN {
```

Comment:
I am wondering if we should the appending of proof at a common place, as right now this code is duplicated between rpc and rest. 
Ideal place for this would be to be part of waku_relay, as essentially RLN is enabled only for relay.

Reply:
Perhaps. Having it as part of waku_relay tightly couples RLN to Relay which might not be ideal, specially considering that lightpush can be used to send messages as well and these messages must have proofs for them to be considered valid

Reply:
Got it, is there any other way to avoid this duplication? 
If not, then fine.

Reply:
Yeah, I'll try to extract it to a function that can be reused in both RPC and REST. Might be an overkill since RPC will die eventually but code will look cleaner anyway